lib/wasi/src/syscalls/mod.rs

```diff
@@ -2659,18 +2659,12 @@ pub fn path_rename<M: MemorySize>(
         source_path.to_str().as_ref().unwrap(),
         true
     ));
-    if state
-        .fs
-        .get_inode_at_path(
-            inodes.deref_mut(),
-            new_fd,
-            target_path.to_str().as_ref().unwrap(),
-            true,
-        )
-        .is_ok()
-    {
-        return Errno::Exist;
-    }
+    state.fs.get_inode_at_path(
+        inodes.deref_mut(),
```

Comment:
We can avoid this (and needing to import `DerefMut`) by taking a mutable reference and letting the compiler auto-deref.

```suggestion
        &mut inodes,
```

Reply:
We use `deref_mut()` in other places within the same file, so this change is probably not going to help.